File: 240225/요일 맞추기/guess-day-of-week.java
import java.io.*;
import java.util.*; 

public class Main {
    static int[] numOfDays = new int[]{0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    static String[] yoils = new String[]{"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
    static BufferedReader buffer = new BufferedReader(new InputStreamReader(System.in));


    static int getTotalDays(int month, int date){
        int totalDays = 0;

        for(int m=1; m<month; m++){
            totalDays+=numOfDays[m];
        }

        return totalDays+date; 
    }


    public static void main(String[] args)throws IOException {
        StringTokenizer tokens = new StringTokenizer(buffer.readLine()); 

        int m1 = Integer.valueOf(tokens.nextToken());
        int d1 = Integer.valueOf(tokens.nextToken());

        int m2 = Integer.valueOf(tokens.nextToken());
        int d2 = Integer.valueOf(tokens.nextToken()); 

        int offset = getTotalDays(m2, d2) - getTotalDays(m1,d1); 
        
    
        int yoilData = offset%7; 
        

        if(yoilData<0){
            yoilData += 7; 
        }


        System.out.println(yoils[yoilData]);



    }
}

File: 250613/요일 맞추기/guess-day-of-week.cpp
#include <iostream>
#include <string> 
using namespace std;

int m1, d1, m2, d2;
int offset = 7777; 
string weekdays[] = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
int days[] ={0, 31,28, 31, 30,31, 30,31,31,30,31,30,31}; 

int getDayCnt(int m, int d){
    int result = 0; 
    for(int month=1; month<m; month++){
        result += days[month]; 
    }

    return result + d; 
}


int main() {
    cin >> m1 >> d1 >> m2 >> d2;

    // 날짜 차이를 구한다 
    int interval = getDayCnt(m2, d2)-getDayCnt(m1,d1)+7777; 

    //7로 나눈 몫으로 요일을 정한다. 
    cout<<weekdays[interval%7];
    return 0;
}